andela/Ah-backend-xmen
# #162725262 Users should be able to like or dislike articles

#### What does this PR do?
This PR adds functionality for articles to be liked or disliked by both the author and readers. And in case you've already liked/disliked an article then you can unlike or un-dislike the article. And finally, you view who liked or disliked.

#### Description of Task to be completed?
##### Modify fetch articles endpoint to have 
```
{
        "likes_count": <number of likes>,
        "dislikes_count": <number of dislikes>
}
```
##### Implement three endpoints
- Like or Dislike an article
`PUT/api/articles/<slug>/likes`
- Fetch the list of users that like or disliked the article
`GET/api/articles/<slug>/likes`
- Remove the like or dislike
`DELETE/api/articles/<slug>/likes`

#### How should this be manually tested?
- First things first, fetch this branch by running the command `git fetch origin ft-like-or-dislike-articles-162725262` in the terminal.
- Check into this branch by running `git checkout ft-like-or-dislike-articles-162725262`
- Activate your virtual environment by running `source venv/bin/activate`
- Apply the new changes to the models by running `python manage.py migrate`
- Drop your current database and create a new one by running `create database <database name>;` and `drop database <database name>;` in the psql shell
- Start the application by running `python manage.py runserver`
- Click the `Run in postman` button to test the endpoint as shown in the screenshots below

#### What are the relevant pivotal tracker stories?
[#162725262](https://www.pivotaltracker.com/story/show/162725262)
#### Screenshots
 
#### LIKE
![image](https://user-images.githubusercontent.com/36506567/51673533-465a4580-1fdf-11e9-8e18-df73056025cf.png)

#### UNLIKE
![image](https://user-images.githubusercontent.com/36506567/51673495-304c8500-1fdf-11e9-802c-82280c06d0c6.png)

#### DISLIKE
![image](https://user-images.githubusercontent.com/36506567/51673528-3f333780-1fdf-11e9-9360-802630992990.png)

#### UN-DISLIKE
![image](https://user-images.githubusercontent.com/36506567/51673473-26c31d00-1fdf-11e9-856b-58723005a0d8.png)

#### LIKES_COUNT
![image](https://user-images.githubusercontent.com/36506567/51527028-b41e3a00-1e44-11e9-927c-bd2a8a625821.png)

#### LIST OF USERS THAT LIKED & DISLIKED
![image](https://user-images.githubusercontent.com/36506567/51673605-7275c680-1fdf-11e9-858c-ed0865bfe5e0.png)

#### POSTMAN
[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/caea5289beb8511bc886)
